File: fin/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height = get_int("How tall is the pyramid? ");
    int spaces;
    string block = "#";
    int blocksonthisrow = 1;
    int blockstobuild;

    while (height < 1 || height > 8) 
        height = get_int("The pyramid must be 8 blocks tall or shorter! How tall is it? ");

    while (height >= 1 && height <= 8) { // A loop to make a new line after each line is created by other loops.

        for (spaces = height - 1; spaces > 0; spaces--) // Calculate and print spaces needed on left side of pyramid.
            printf(" ");

        for (blockstobuild = blocksonthisrow; blockstobuild > 0; blockstobuild--) // Calculate and print blocks needed on left side.
            printf("%s", block);

        printf("  "); // Space out left and right blocks.

        for (blockstobuild = blocksonthisrow; blockstobuild >= 1; blockstobuild--) // Calculate and print blocks needed on right side. Repetative; better method?
            printf("%s", block);

        printf("\n");

        height--;
        blocksonthisrow++;
    }
}

File: fin/sentimental-mario-more/mario.py
# TODO
from cs50 import get_int

while True:
    height = get_int("Height (between 1 and 8): ")
    if height > 0 and height < 9:
        break;

for h in range(height):
    print((" " * (height - h - 1)), end="")
    print("#" * (h + 1), end="")
    print("  ", end="")
    print("#" * (h + 1))
